### PR TITLE
Test re-enabling osx travis

### PR DESCRIPTION
**Pull request by tkelman**
Following up on #11614, we might be able to get away with turning osx Travis back on now that we're running the tests with a release build of julia instead of debug. Also some other grab-bag cleanup of Travis:

Add local.ji test with julia-debug

Don't build rmath on Travis, it's unused in base (the way in which I'm doing this via `STAGE2_DEPS` is a little hacky, I'll admit - we have no other way of turning off the rmath build right now, aside from disabling gpl libraries completely)

Cleanup make flags and apt-get packages for zlib and rmath that don't do anything

Add timestamps to make install step

Use parallel make

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,37 +31,39 @@ before_install:
...
         export LDFLAGS="-L$(brew --prefix openblas-julia)/lib -L$(brew --prefix suite-sparse-julia)/lib";
         export DYLD_FALLBACK_LIBRARY_PATH="/usr/local/lib:/lib:/usr/lib:$(brew --prefix openblas-julia)/lib:$(brew --prefix suite-sparse-julia)/lib:$(brew --prefix arpack-julia)/lib";
         make $BUILDOPTS -C contrib -f repackage_system_suitesparse4.make;
       fi
+    - git clone -q git://git.kitenet.net/moreutils
 script:
     - make check-whitespace || exit 1
     - make $BUILDOPTS -C base version_git.jl.phony
-    - make $BUILDOPTS NO_GIT=1 prefix=/tmp/julia install
+    - make $BUILDOPTS NO_GIT=1 prefix=/tmp/julia install | moreutils/ts -s "%.s"
```
Comment (tkelman) [inline .travis.yml]: blatantly ripped off from https://github.com/mlubin/JuMPSupplement/blob/master/acpower/README.md, thanks for the idea @mlubin

Comment (IainNZ) [inline .travis.yml]: That `ts` really made life quite nice

Comment (tkelman) [inline .travis.yml]: annoyingly the `-s` relative-to-start-time flag isn't there yet in the versions packaged in ubuntu, but at least it's a just perl script and can be run from a git clone of the source

**Comment by tkelman** [comment]
Thoughts, anybody?

**Comment by yuyichao** [comment]
Maybe removing `USE_SYSTEM_RMATH` completely ? [(i.e. here)](https://github.com/JuliaLang/julia/blob/1c69d6324711d03deeb2d470c5c4f0d4017eae71/contrib/mac/macports.make#L32)

**Comment by tkelman** [comment]
thought about it, but didn't feel like touching macports (should we just delete that?)

**Comment by yuyichao** [comment]
That's what I thought, unless there's sth subtle here...

(I guess this is not what this PR is for. I just saw the removal of the option and found another use of it with `git grep`.)

**Comment by ViralBShah** [comment]
Perhaps Distribution.jl testing may fail if we do not build RMath. We should just put it into BinDeps reliably.

**Comment by yuyichao** [comment]
@ViralBShah I think this is not about disabling RMath. It just seems to be an unused make option from the output below.

```
yuyichao% git grep USE_SYSTEM_RMATH | cat
Make.inc:USE_SYSTEM_RMATH=0
contrib/mac/macports.make:              USE_SYSTEM_RMATH=0 \
```

**Comment by tkelman** [comment]
`USE_SYSTEM_RMATH` doesn't actually work so I'm taking it out of `Make.inc`, and I'm not building it on Travis since nothing on Travis tests it. This doesn't change the fact that the library is still getting built into the binaries from the buildbots, that will still be the case for now but it would be great to transition rmath over to using bindeps instead of being part of base for 0.5.

**Comment by tkelman** [comment]
I'm gonna merge, testing on more platforms is a good thing. We can comment it back out if it still times out regularly.

**Pull request merged**